test.ts

```diff
@@ -131,12 +129,13 @@ export const findResourceInstance = <T extends "coder_script" | string>(
   return resource.instances[0].attributes as any;
 };
 
-// testRequiredVariables creates a test-case
-// for each variable provided and ensures that
-// the apply fails without it.
-export const testRequiredVariables = (
+/**
+ * Creates a test-case for each variable provided and ensures that the apply
+ * fails without it.
+ */
+export const testRequiredVariables = <TVars extends Record<string, string>>(
```

Comment:
Added a type parameter so that you can define what properties you expect to be valid for the function, so that you can more easily catch accidental typos